### PR TITLE
#8737 Package standalone mods on save - Slice 1

### LINKED ISSUE #8737: Slice 1: Show create-mod modal on save of standalone mod components
Sketch here: https://www.notion.so/pixiebrix/Page-Editor-always-package-standalone-mods-on-save-e9d8189618944e4e9ee1edc292fd1796?pvs=4#b75cfd5eab3147f1a0f723691d757a65

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Closes #8737 

## Discussion

- I removed the `widerLabel` thing because it was making the create mod modal look bad, too much white space and the user scope input squished. I went through other usages and they were all super weird and/or didn't seem to affect the UI at all.

## Remaining Work
- [ ] Fix issue(s) identified by the end-to-end tests
- [ ] Fix any remaining merge conflicts

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/pageEditor/modListingPanel/DraftModComponentListItem.tsx [hunk 1]
```diff
@@ -128,13 +123,11 @@ const DraftModComponentListItem: React.FunctionComponent<
...
     } else {
-      await saveStandaloneModComponent(modComponentFormState);
+      dispatch(actions.showCreateModModal({ keepLocalCopy: false }));
     }
   };
 
-  const isSaving = modComponentFormState.modMetadata
-    ? isSavingMod
-    : isSavingStandaloneModComponent;
+  const isSaving = modComponentFormState.modMetadata ? isSavingMod : false;
```
Comment (BLoe) [inline src/pageEditor/modListingPanel/DraftModComponentListItem.tsx:130]: This is only used to disable the action menu during save, but when a modal is open, the modal backdrop prevents clicking on anything anyway, so we don't really need to care about this any longer for mod components.

File: src/pageEditor/hooks/useCreateModFromModComponent.ts [hunk 2]
```diff
@@ -107,9 +112,21 @@ function useCreateModFromModComponent(
...
+              //  the "keep local copy" option when converting to a mod.
+              //  Also since v2.0.5, saving a standalone mod component in the page editor will call
+              //  this create hook with keepLocalCopy forced to false.
+              deleteStandaloneModDefinition({
+                extensionId: activeModComponent.uuid,
+              }),
+              removeModComponentFromStorage({
+                modComponentId: activeModComponent.uuid,
+              }),
+            ]);
```
Comment (grahamlangford) [inline src/pageEditor/hooks/useCreateModFromModComponent.ts:129]: 💜

File: src/pageEditor/modListingPanel/DraftModComponentListItem.tsx (hunk 1 above)
Comment (grahamlangford) [reply src/pageEditor/modListingPanel/DraftModComponentListItem.tsx:130]: In that case, why don't we remove it entirely?

**Review by grahamlangford — APPROVED** [review]

**Comment by twschiller** [comment]
@BLoe see merge conflicts and e2e tests

File: src/pageEditor/modListingPanel/DraftModComponentListItem.tsx (hunk 1 above)
Comment (BLoe) [reply src/pageEditor/modListingPanel/DraftModComponentListItem.tsx:130]: I was leaving it in for the save-mod case, where we aren't showing a modal or anything

**Pull request merged**